packages/remix-testing/create-remix-stub.tsx

```diff
@@ -0,0 +1,319 @@
+): RouteManifest<EntryRoute> {
+  return routes.reduce((manifest, route) => {
+    if (route.children) {
+      createRouteManifest(route.children, manifest, route.id);
+    }
+    manifest[route.id!] = convertToEntryRoute(route, parentId);
+    return manifest;
+  }, manifest || {});
+}
+
+function createRouteModules(
+  routes: MockRouteObject[],
+  routeModules?: RouteModules
+): RouteModules {
+  return routes.reduce((modules, route) => {
+    if (route.children) {
+      createRouteModules(route.children, modules);
+    }
+
+    if (typeof route.id === "undefined") {
+      throw new Error("Route id must be defined");
+    }
+
+    modules[route.id] = {
+      CatchBoundary: route.CatchBoundary,
+      ErrorBoundary: route.ErrorBoundary,
+      default: () => <>{route.element}</>,
+      handle: route.handle,
+      links: route.links,
+      meta: route.meta,
+      unstable_shouldReload: route.unstable_shouldReload,
+    };
+    return modules;
+  }, routeModules || {});
+}
+
+const originalFetch =
+  typeof global !== "undefined" ? global.fetch : window.fetch;
+
+function monkeyPatchFetch(
+  queryRoute: StaticHandler["queryRoute"],
+  dataRoutes: StaticHandler["dataRoutes"]
+) {
+  let fetchPatch = async (
+    input: RequestInfo | URL,
+    init: RequestInit = {}
+  ): Promise<Response> => {
+    let request = new Request(input, init);
+    let url = new URL(request.url);
+
+    // if we have matches, send the request to mock routes via @remix-run/router rather than the normal
+    // @remix-run/server-runtime so that stubs can also be used in browser environments.
+    let matches = matchRoutes(dataRoutes, url);
+    if (matches && matchRoutes.length > 0) {
+      let response = await queryRoute(request);
+
+      if (response instanceof Response) {
+        return response;
+      }
+
+      return json(response);
+    }
+
+    // if no matches, passthrough to the original fetch as mock routes couldn't handle the request.
+    return originalFetch(request, init);
+  };
+
+  if (typeof global !== "undefined") {
+    global.fetch = fetchPatch;
+  } else {
+    window.fetch = fetchPatch;
+  }
+}
+
+function convertToEntryRoute(
+  route: MockRouteObject,
+  parentId?: string
+): EntryRoute {
+  return {
+    id: route.id!,
+    index: route.index,
+    caseSensitive: route.caseSensitive,
+    path: route.path,
+    parentId,
+    hasAction: !!route.action,
+    hasLoader: !!route.loader,
+    module: "",
+    hasCatchBoundary: !!route.CatchBoundary,
+    hasErrorBoundary: !!route.ErrorBoundary,
+  };
+}
+
+function convertToEntryRouteMatch(
+  routes: AgnosticRouteMatch<string, MockRouteObject>[]
+) {
+  return routes.map((match) => {
+    return {
+      params: match.params,
+      pathname: match.pathname,
+      route: convertToEntryRoute(match.route),
+    };
+  });
+}
+
+// Converts route data from a path based index to a route id index value.
+// e.g. { "/post/:postId": post } to { "0": post }
+function convertRouteData(
```

Comment:
I've updated this function in remix stubs to [support nested routes](https://github.com/jrestall/remix-stubs/blob/7eca61c202cf1eff29b7060b55d12f6586f437d7/packages/remix-stubs/src/create-remix-stub.tsx#L308). It'll need a similar change here. I also  used `'/'` to refer to route data for the pathless root route which I think is accurate but you guys will know best.